Lib/test/test_httpservers.py

```diff
@@ -1536,6 +1541,254 @@ def test_server_test_ipv4(self, _):
+                                  tls_key_option, self.tls_key)
+                self.args['tls_cert'] = self.tls_cert
+                self.args['tls_key'] = self.tls_key
+                mock_func.assert_called_once_with(**self.args)
+                self.args['tls_cert'] = None
+                self.args['tls_key'] = None
+                mock_func.reset_mock()
+
+        # Normal: --tls-cert, --tls-key and --tls-password-file
+
+        for tls_cert_option in tls_cert_options:
+            for tls_key_option in tls_key_options:
+                for tls_password_option in tls_password_options:
+                    self.invoke_httpd(tls_cert_option,
+                                      self.tls_cert,
+                                      tls_key_option,
+                                      self.tls_key,
+                                      tls_password_option,
+                                      self.tls_password_file)
+                    self.args['tls_cert'] = self.tls_cert
+                    self.args['tls_key'] = self.tls_key
+                    self.args['tls_password'] = self.tls_password
+                    mock_func.assert_called_once_with(**self.args)
+                    self.args['tls_cert'] = None
+                    self.args['tls_key'] = None
+                    self.args['tls_password'] = None
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-key without --tls-cert
+
+        for tls_key_option in tls_key_options:
+            for tls_cert_option in tls_cert_options:
+                with self.assertRaises(SystemExit):
+                    self.invoke_httpd(tls_key_option, self.tls_key)
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file without --tls-cert
+
+        for tls_password_option in tls_password_options:
+            with self.assertRaises(SystemExit):
+                self.invoke_httpd(tls_password_option, self.tls_password_file)
+                mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file cannot be opened
+
+        non_existent_file = 'non_existent_file'
+        for tls_password_option in tls_password_options:
+            for tls_cert_option in tls_cert_options:
+                with self.assertRaises(SystemExit):
+                    self.invoke_httpd(tls_cert_option,
+                                      self.tls_cert,
+                                      tls_password_option,
+                                      non_existent_file)
+
+    @mock.patch('http.server.test')
+    def test_no_arguments(self, mock_func):
+        self.invoke_httpd()
+        mock_func.assert_called_once_with(**self.args)
+        mock_func.reset_mock()
+
+    @mock.patch('http.server.test')
+    def test_help_flag(self, _):
+        options = ['-h', '--help']
+        for option in options:
+            with self.assertRaises(SystemExit):
+                output = self.invoke_httpd(option)
```

Comment:
Actually, this function never returns as it raises SystemExit. I don't think you actually need to check for the output. If you want to check for it, you need to redirect stderr/stdout before calling `invoke_httpd` actually.